sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodCheck.java

```diff
@@ -0,0 +1,258 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.github.sevntu.checkstyle.Utils;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Check that forbidden method is not used. We can forbid a method by method name and number of
+ * arguments.
+ * This can be used to enforce things like:
+ * <ul>
+ * <li> System.exit() should not be called.</li>
+ * <li> Assert.assertTrue() and Assert.assertFalse() have a 1 arg variant that does not provide a
+ * helpful message on failure.
+ * </ul>
+ * Parameters are:
+ * <ul>
+ * <li><b>methodName</b> - Regex to match name of the method to be forbidden.
+ * When blank or unspecified, all the methods are allowed.</li>
+ * <li><b>argumentCount</b> - Range to match number of arguments the method takes.
```

Comment:
It is positive integer or range, as in most cases it will be user by simple number. 
Please fix documentation.

Reply:
Zero is also acceptable value so I have used Number or Range.